src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -2060,22 +2059,19 @@ private AssemblySymbol GetForwardedToAssembly(string fullName, int arity, Diagno
             return null;
         }
 
-        internal static void CheckFeatureAvailability(Location location, MessageID feature, DiagnosticBag diagnostics)
+        internal static void CheckFeatureAvailability(SyntaxNode syntax, MessageID feature, DiagnosticBag diagnostics, Location locationOpt = null)
         {
-            var options = (CSharpParseOptions)location.SourceTree.Options;
+            var options = (CSharpParseOptions)syntax.SyntaxTree.Options;
```

Comment:
It looks like speculative SemanticModel never passes diagnostics to its clients. Given that, it really doesn't matter what options we use here because they only affect whether an error is reported or not.